osquery/tables/system/darwin/processes.cpp

```diff
@@ -346,26 +370,28 @@ QueryData genProcesses(QueryContext& context) {
           static_cast<long>(diff * multiply) / NSECS_IN_USEC;
 
       // Get the start_time of process since the computer started
-      r["start_time"] = TEXT((uptime + seconds_since_launch) / CPU_TIME_RATIO);
+      r.start_time = ((uptime + seconds_since_launch) / CPU_TIME_RATIO);
     } else {
-      r["wired_size"] = "-1";
-      r["resident_size"] = "-1";
-      r["total_size"] = "-1";
-      r["user_time"] = "-1";
-      r["system_time"] = "-1";
-      r["start_time"] = "-1";
+      r.wired_size = -1;
+      r.resident_size = -1;
+      r.total_size = -1;
+      r.user_time = -1;
+      r.system_time = -1;
+      r.start_time = -1;
     }
 
     struct proc_taskinfo task_info;
     status =
         proc_pidinfo(pid, PROC_PIDTASKINFO, 0, &task_info, sizeof(task_info));
     if (status == sizeof(task_info)) {
-      r["threads"] = INTEGER(task_info.pti_threadnum);
+      r.threads = (task_info.pti_threadnum);
     } else {
-      r["threads"] = "-1";
+      r.threads = -1;
     }
 
-    results.push_back(r);
+    Row row;
+    r.toHashmap(row);
```

Comment:
cc @theopolis 

This is interesting! Using a structure generated by the table schema @packetzero has gotten the compiler to verify that rows are being created with respect to their definition. Are there any potential reasons consequences of doing it this way that you can think of, or should we eventually code mod the rest of the tables to be this way?

Reply:
Actually, the exciting part is the follow-on work in a different branch (typed_rows), where table plugins can return a row struct without the need for column names or marshaling types to string before passing to sqlite.